utils/query_engine/dual_qdrant_retrieval_engine.py

```
@@ -26,13 +32,67 @@ class DualQdrantRetrievalEngine(CustomQueryEngine):
+            # Take top-k nodes and update their scores
+            top_k = min(RERANK_TOP_K, len(scored_nodes))
+            reranked_nodes = []
+            
+            for i in range(top_k):
+                node, relevance_score = scored_nodes[i]
+                # Update the score with the reranking score
+                node.score = float(relevance_score)
+                reranked_nodes.append(node)
+                
+            logging.info(f"Reranked {len(nodes)} nodes to top {len(reranked_nodes)} nodes using CrossEncoder")
+            return reranked_nodes
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)